sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1047,7 +1047,8 @@ trait ArraySortLike extends ExpectsInputTypes {
         s"int $c = ${ctx.genComp(elementType, s"(($jt) $o1)", s"(($jt) $o2)")};"
       }
       val nonNullPrimitiveAscendingSort =
-        if (CodeGenerator.isPrimitiveType(elementType) && !containsNull) {
+        if (CodeGenerator.isPrimitiveType(elementType) && elementType != BooleanType
+          && !containsNull) {
```

Comment:
How about making this condition a separate method like `isFastSortSupported`?

Reply:
will make a change